src/azure-cli-core/azure/cli/core/auth/msal_authentication.py

```diff
@@ -38,17 +38,20 @@ def __init__(self, client_id, username, **kwargs):
         """
         super().__init__(client_id, **kwargs)
 
-        accounts = self.get_accounts(username)
+        # Make sure username is specified, otherwise MSAL returns all accounts
+        assert username, "username must be specified, got {!r}".format(username)
```

Comment:
We don't raise `CLIError` because username should never happen. When that indeed happens, it is a bug.